lib/base/utility.hpp

```diff
@@ -80,6 +89,14 @@ class Utility
 #ifndef _WIN32
 	static void SetNonBlocking(int fd, bool nb = true);
 	static void SetCloExec(int fd, bool cloexec = true);
+
+	template<size_t exceptAmount>
+	static inline
+	void CloseAllFDs(const std::array<int, exceptAmount>& except);
+
+	template<size_t exceptAmount, class OnClose>
+	static inline
+	void CloseAllFDs(const std::array<int, exceptAmount>& except, OnClose onClose);
```

Comment:
Is there any real advantage of using `std::array` over `std::vector`? With vector, one could avoid templating this altogether. Yes, this probably means there will be an additional memory allocation, but after all, this is done after a fork, so that's cheap in comparison.

Reply:
1. Yes, I'm just avoiding malloc(3). But if I can avoid malloc(3), I avoid malloc(3). (https://github.com/Icinga/docker-icinga2/pull/22#issuecomment-669802604)
2. No, I'd avoid just one of two template params.